src/lib/common/string.cpp

```diff
@@ -774,11 +750,12 @@ std::string servicePathCheck(const char* servicePath)
 * It IS a double, but this function will say it is not, as it actually is not a valid double for
 * this computer as the computer cannot represent it as the C builtin type 'double' ... OK!
 */
-bool str2double(char* s, double* dP)
+bool str2double(const char* s, double* dP)
 {
-  char*   rest = s;
+  char*   rest = (char*) s;
```

Comment:
AFAICS, is an out (write-only) parameter for ```strtod```, the initial value should not be taken into account and the value after the call is what is needed/used in the following statements